bentoml_cli/worker/http_dev_api_server.py

```diff
@@ -122,13 +122,13 @@ def main(
 
         if not ssl_version:
             ssl_version = ssl.PROTOCOL_TLS_SERVER
-            uvicorn_options["ssl_version"] = ssl_version
+        uvicorn_options["ssl_version"] = int(ssl_version)
```

Comment:
Shouldn't we just change the argument types? Also, technically I think this is unrelated to gRPC :P

Reply:
ssl_version is int value, whereas the types from click is string

the reason of click types is string is that circus args only takes string, and if we give it int, circus would simply fail.

So we have to convert all of that to string and then convert again back to its original types.

Reply:
I'm not entirely sure what you mean by "circus args only takes string", but all arguments are strings, surely? The click types are just utilities that convert string to whatever type you want, right?

Reply:
https://github.com/bentoml/BentoML/blob/e2910399adb5a8ea19317b4fb54390da2d265408/bentoml/serve.py#L151

It has to do with this, if you pass the args value other than string, then circus will fail.

Uvicorn options doesn't accept strings for these field, hence the conversion.

Reply:
I mean you should change the circus arguments to ints; *all* program arguments are strings. Click should do the parsing for us.

Reply:
Wait, `ssl_version` and `ssl_cert_reqs` are already int? You just need to change the type annotations here.

Reply:
click doesn't convert types.